acme/tls_alpn_challenge_test.go

```diff
@@ -0,0 +1,87 @@
+package acme
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/subtle"
+	"crypto/tls"
+	"strings"
+	"testing"
+)
+
+func TestTLSALPNChallenge(t *testing.T) {
+	domain := "localhost:23457"
+	privKey, _ := rsa.GenerateKey(rand.Reader, 512)
+	j := &jws{privKey: privKey}
+	clientChallenge := challenge{Type: string(TLSALPN01), Token: "tlsalpn1"}
+	mockValidate := func(_ *jws, _, _ string, chlng challenge) error {
+		conn, err := tls.Dial("tcp", domain, &tls.Config{
+			InsecureSkipVerify: true,
+		})
+		if err != nil {
+			t.Errorf("Expected to connect to challenge server without an error. %s", err.Error())
```

Comment:
```go
t.Errorf("Expected to connect to challenge server without an error. %v", err)

Reply:
Why is this preferred?

Reply:
In case the error value is not a Stringer. Like, if it is a regular struct. It's just more robust.

Reply:
I can make the change, for sure, but `error` is just an interface for:

```go
type error interface {
  Error() string
}
```

So technically, every error, when called with `.Error()` will return a string, else it wouldn't compile.